lib/DiscoveryStreamFeed.jsm

```diff
@@ -412,6 +415,26 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
     this.loaded = true;
   }
 
+  // We have to rotate stories on the client so that
+  // active stories are at the front of the list, followed by stories that have expired
+  // impressions i.e. have been displayed for longer than recsExpireTime.
+  rotate(feedResponse) {
+    const {recommendations} = feedResponse;
+
+    const maxImpressionAge = Math.max(feedResponse.settings.recsExpireTime * 1000 || DEFAULT_RECS_EXPIRE_TIME, DEFAULT_RECS_EXPIRE_TIME);
+    const impressions = this.readImpressionsPref(PREF_REC_IMPRESSIONS);
+    const expired = [];
+    const active = [];
+    for (const item of recommendations) {
+      if (impressions[item.id] && Date.now() - impressions[item.id] >= maxImpressionAge) {
```

Comment:
What's the expected behavior? If a story was most recently seen more than 1 hour ago, expire it? Or should it be story was first seen 1 hour ago, expire it?

Reply:
Checking, you may have found a bug.

Reply:
It should be first seen more than x time again, expire it.